src/pages/index.js

```diff
@@ -29,60 +30,155 @@ const enableValidation = (config) => {
 
 enableValidation(validationConfig);
 
-//creating cards from the array
-const popupWithImage = new PopupWithImage(".modal_type_show-image");
+//api
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "9e620f98-b2bc-4f4a-81bd-341fb1cf797f",
+  },
+});
 
-const handleImageClick = (card) => {
-  popupWithImage.open(card.link, card.name);
+//user
+api
+  .getUserInfo()
+  .then((userInfo) => {
+    //set user info
+    const user = new UserInfo(
+      ".explorer__name",
+      ".explorer__description",
+      ".explorer__photo"
+    );
+    user.setUserInfo({
+      name: userInfo.name,
+      about: userInfo.about,
+    });
+    user.setAvatar({ avatar: userInfo.avatar });
+    return user;
+  })
+  .then((user) => {
+    //cahnging user details
+    function saveProfileChanges(newUserData) {
+      profileEditPopup.savingDisplay(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```